app/controllers/registration_controller.rb

```diff
@@ -5,14 +5,42 @@ def index
 
   def create
     @user = User.new(user_params)
+    @user.token = token(@user.email)
     return render 'index' unless @user.save
     session[:user_id] = @user.id
     redirect_to home_index_path
   end
 
+  def confirm
+    @user = User.find_by(token: params[:token])
+    if @user
+      @user.active = true
+      return render 'error' unless @user.save
+      redirect_to home_index_path
+    else
+      render 'error'
+    end
+  end
+
+  def signup_email
+    user = User.find(session[:user_id])
+    user.token = token(user.email)
+    return render 'error' unless user.save
+    send_email
```

Comment:
what is this doing?

Reply:
I'm finding the user that is logged but not active and changing his token, maybe i can implement this action on the model